overlord/servicestate/servicemgr.go

```diff
@@ -117,6 +117,34 @@ func (m *ServiceManager) ensureSnapServicesUpdated() (err error) {
 		return nil
 	}
 
+	var sysd systemd.Systemd
+	if len(allStates) > 0 {
+		if snapdenv.Preseeding() {
+			sysd = systemd.NewEmulationMode(dirs.GlobalRootDir)
+		} else {
+			sysd = systemd.New(systemd.SystemMode, nil)
+		}
+	}
+
+	for _, snapSt := range allStates {
+		info, err := snapSt.CurrentInfo()
+		if err != nil {
+			return err
+		}
+		squashfsPath := dirs.StripRootDir(info.MountFile())
+		whereDir := dirs.StripRootDir(info.MountDir())
+		_, err = sysd.EnsureMountUnitFile(
+			info.InstanceName(),
+			info.Revision.String(),
+			squashfsPath,
+			whereDir,
+			"squashfs",
+		)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
the changes made to servicemgr_test.go don't seem to care if the code is here or removed

Reply:
I have added 3 tests.